src/pandantic/validators/pandas.py

```diff
@@ -0,0 +1,136 @@
+import logging
+import math
+import os
+from typing import Any, Optional
+
+import pandas as pd
+from multiprocess import (  # type:ignore # pylint: disable=no-name-in-module
+    Process,
+    Queue,
+)
+
+from pandantic.types import SchemaTypes
+from pandantic.validators.baseclass import BaseValidator
+
+
+class PandasValidator(BaseValidator):
+    def __init__(self, schema: SchemaTypes):
+        self.schema = schema
+
+    def validate(
+        self,
+        dataframe: pd.DataFrame,
+        errors: str = "raise",
```

Comment:
Kinda a fun idea, but what about providing type safety for the functions using [`pydantic` decorators](https://docs.pydantic.dev/latest/concepts/validation_decorator/)? I've never used them before, but it could be cool as it double-down on the whole "pydantic" ecosystem/pattern you see in `FastAPI` and other

Reply:
This actually only works with more primitive input types. Nevermind!